src/components/ReturnsCalculator.jsx

```diff
@@ -0,0 +1,244 @@
+import React           from "react";
+import { Link, Route } from "react-router-dom";
+import {
+	Box,
+	Heading,
+	Text,
+	Flex,
+	InputGroup,
+	Input,
+	InputRightAddon,
+	Button,
+	useColorMode,
+	Slider,
+	SliderTrack,
+	SliderFilledTrack,
+	SliderThumb
+}                      from "@chakra-ui/core";
+import CustomButton    from "./CustomButton";
+import CountUp         from "react-countup";
+import Helmet          from "react-helmet";
+import { useDebounce } from "use-debounce";
+import LogEvent        from "./LogEvent";
+import ErrorMessage    from "./ErrorMessage";
+import ExpectedReturns from "./SuggestedValidators/ExpectedReturns";
+
+const textColor = { light: "gray.600", dark: "#FFF" };
+
+type ReturnsCalculatorProps = {
+	colorMode: "light" | "dark",
+	validatorData: Array<{}>,
+};
+
+const ReturnsCalculator = (props: ReturnsCalculatorProps) => {
+	const { colorMode, toggleColorMode }        = useColorMode();
+	const [stakeInput, setStakeInput]           = React.useState();
+	const [expectedReturns, setExpectedReturns] = React.useState(0.0);
+	const [suggPromptsAmount]                   = useDebounce(stakeInput / 16, 0);
+	const [validatorsList, setValidatorsList]   = React.useState([]);
+	const [validatorData, setValidatorData]     = React.useState([]);
+	const [errorState, setErrorState]           = React.useState(false);
+	const [intentionData, setIntentionData]     = React.useState([]);
+	const [apiConnected, setApiConnected]       = React.useState(false);
+	const [isLoaded, setIsLoaded]               = React.useState(false);
+	const [riskLevel, setRiskLevel]             = React.useState(50);
+	const [sliderBG, setSliderBG]               = React.useState("yellow.300");
+	const ERA_PER_DAY = 4;
+	// console.log('props - ', props.validatorData);
+
+	function suggPrompts() {
+		const data = validatorsList.map(validator => {
+			const {
+				stashId,
+				stashIdTruncated,
+				name,
+				commission,
+				totalStake,
+				poolReward,
+				noOfNominators
+			} = validator;
+			const userStakeFraction =
+				suggPromptsAmount / (suggPromptsAmount + totalStake);
+			const dailyEarning = userStakeFraction * poolReward * ERA_PER_DAY;
+			return {
+				noOfNominators,
+				stashId,
+				stashIdTruncated,
+				name,
+				commission: `${parseFloat(commission)}%`,
+				dailyEarning: isNaN(dailyEarning)
+					? "Not enough data"
+					: `${dailyEarning.toPrecision(10)} KSM`,
+				dailyEarningPrecise: isNaN(dailyEarning) ? 0 : dailyEarning
+			};
+		});
+
+		//handle top 16 suggested validaors
+		data.sort((a, b) => b.dailyEarningPrecise - a.dailyEarningPrecise);
+		const top16data = [...data.slice(0, 16)];
```

Comment:
Better name required -> for example `suggestedValidators` or `suggestedValidatorsInfo`

Reply:
changed to suggested validators.